src/main/java/seedu/address/logic/CommandHistory.java

```diff
@@ -0,0 +1,65 @@
+package seedu.address.logic;
+
+import java.util.ArrayList;
+
+/**
+ * Represents an abstraction for a command history data structure.
+ */
+public class CommandHistory {
+
+    private final ArrayList<String> commandHistory = new ArrayList<>();
+    private int index = 0;
+
+    /**
+     * Get the previous command text.
+     *
+     * @return the previous command text.
+     */
+    public String getPrevious() {
+        assert -1 <= index && index <= commandHistory.size();
+
+        if (index < 0) {
+            return "";
+        }
+
+        index -= 1;
+
+        if (index < 0) {
+            return "";
+        }
+
+        return commandHistory.get(index);
```

Comment:
Would it be better to change this to:
```
assert 0 <= index && index <= commandHistory.size();

if (commandHistory.isEmpty()) {
    return "";
}

index = Math.max(0, index - 1);

return commandHistory.get(index);
```
instead? This will also follow the default terminal logic of remaining at the oldest command if there are no older commands.

Reply:
Code logic has been changed to throw an exception.